tests/tt_metal/tt_metal/unit_tests/basic/test_noc.cpp

```diff
@@ -97,7 +97,7 @@ TEST_F(BasicFixture, VerifyNocNodeIDs) {
     ASSERT_TRUE(tt::tt_metal::CloseDevice(device));
 }
 TEST_F(BasicFixture, VerifyNocIdentityTranslationTable) {
-    auto arch = tt::get_arch_from_string(get_env_arch_name());
+    auto arch = tt::get_arch_from_string(get_umd_arch_name());
```

Comment:
wonder why do we do this everywhere instead of some simple `tt::get_arch()`

Reply:
You are right to point this out.

Good catch.

The previous function call needed the conversion because an ENV is a string.

but now I can get rid of this

Reply:
On second thought...

The return could possibly becoming from an ENV still.

```
    if(std::getenv("TT_METAL_SIMULATOR_EN")) {
        return get_env_arch_name();
    }
```

I don't know what tests are relevant to real hardware or what tests are relevant to simulation.

So I'll just leave it unoptimized like this.